lib/ioh-zfs

```diff
@@ -2,44 +2,75 @@
 
 # Set ZFS properties
 __zfs_set() {
-	local name="$2"
-	if [ -z $3 ]; then
+	shift 1
+	if [ $# -lt 2 ]; then
 		printf "missing argument\nusage:\n"
-		printf "\tset <name> <property=value> ...\n"
+		printf "\tset <property=value> ... <name>\n"
+		exit 1
+	fi
+
+	# Get last element in the arguemnts
+	local name="$(echo "$@" | rev | cut -d' ' -f1 | rev)"
+	local dataset="$(zfs get -H -s local -o name,value -t filesystem iohyve:name | grep "$(printf '\t')$name$" | cut -f1)"
+
+	if [ -z $dataset ]; then
+		printf "'$name' doesn't exist\n"
 		exit 1
 	fi
-	local pool="$(zfs list -H -t volume | grep iohyve/$name | cut -d '/' -f 1 | head -n1)"
-	shift 2
+
+	local var=0
 	for arg in "$@"; do
-		local prop="$(echo $arg | cut -d '=' -f1)"
-		local val="$(echo $arg | cut -d '=' -f2)"
+		# Increment loop counter
+		var=$(($var+1))
```

Comment:
I do not like the loop counter, but couldn't come up with a better solution.
Any suggestions?